### ISSUE TITLE
News downloading broken, caused error and now causes lack of news data since comic 3777

### ISSUE THREAD

**Issue by trlkly** [Questionable-Content-Extensions/client#24]
[Comic 3777](https://questionablecontent.net/view.php?comic=3777) puts up this giant error in a red overlay that, despite my attempts, I cannot copy and paste. It begins with an error about the call to setComic function being set to null. 

[The previous comic](https://questionablecontent.net/view.php?comic=3776) also doesn't have any data. That may be related. 

Note: if the problem is a lack of data, I would be willing to take on some tagging duties. I understand that you are rather busy.

**Comment by ilyvion** [comment]
Sorry about the issues. I'll have a look at it as soon as I can; possibly
as soon as tonight.

The issue should not be related to a lack of data... Then we'd see this
issue all the time.

**Comment by trlkly** [comment]
True, but I thought maybe the problem was that there are two comics in a row without data. Usually there's only one that gets its data sometime during the day.

**Comment by trlkly** [comment]
UPDATE: No longer getting an error, but also no data is displaying on any comic since (and including) [Comic 3776](https://questionablecontent.net/view.php?comic=3776).

**Comment by ilyvion** [comment]
Hello again.

Sorry about the slow responses, I have some personal issues going on at the moment, and this is the first time I'm on my computer in probably half a month... anyway.

I just quickly fixed the issue by logging on to my server from my phone and hot fixing the problem. The lack of data, though, was probably because my contributors were all otherwise occupied at the time. The data seems to have caught up now. If you're serious about wanting to be a contributor, we can talk about once I'm less caught up in other stuff.

To sum up the issue, something broke in my news parsing code, and the code wasn't written to handle that possibility, so everything broke. For now, news does not get fetched and stored, so the news section will be empty from 3777 onwards until I fix the underlying problem. But everything else should work now.

**Comment by ilyvion** [comment]
Alright, everything is back to normal now. Thanks for the report, and let me know if something else breaks. :)
